### ISSUE TITLE
Support keyboard repetition in terminal

### ISSUE THREAD

**Issue by liebkne** [zed-industries/community#1550]
### Check for existing issues

- [X] Completed

### Describe the feature

Keyboard repetition does not work for some keys like `a-z` in terminal while it works in `Alacritty`.

**Comment by bswinnerton** [comment]
Adding support for this would be particularly helpful for holding the `j` or `k` keys when scrolling through a large command output like `git diff`.

**Comment by Kristoffer** [comment]
This also affects running vi and vim in terminal, which causes arrowkeys (h j k l) to not perform.

**Comment by mrnugget** [comment]
Do you have concrete steps to reproduce? I have Vim mode enabled in Zed and can hold down `j` to scroll down the buffer. I can do the same thing inside the built-in terminal: keep typing `j` in the shell, or scroll in Vim inside that built-in terminal.

https://github.com/zed-industries/community/assets/1185253/1fb0f128-b92f-46a4-9c23-0f289532585e

Do you use this setting too?

```
defaults write -g ApplePressAndHoldEnabled -bool false
```

That disables press-and-hold to type special characters.

(Doesn't mean that users should have to use that to make it work in Zed, but just trying to find out why it works for me and not for you)

**Comment by Kristoffer** [comment]
aha, that works, to disable ApplePressAndHoldEnabled.

So to me it looks like vi and vim in 
- OSX terminal
- iTerm2 
- Zed
- VSCode
bypasses ApplePressAndHoldEnabled when in VI mode.

But the Zed terminal does not.

I see I'm actually using en_DK, from `defaults read -g ` if that has any relevance:

`{
    AKLastLocale = "en_DK";
    AppleInterfaceStyleSwitchesAutomatically = 1;
    AppleKeyboardUIMode = 3;
    AppleLanguages =     (
        "en-DK"
    );
    AppleLanguagesDidMigrate = "13.4.1";
    AppleLanguagesSchemaVersion = 4000;
    AppleLocale = "en_DK";
    AppleMenuBarVisibleInFullscreen = 0;
    AppleMiniaturizeOnDoubleClick = 0;
    ApplePressAndHoldEnabled = 0;
    AppleShowAllExtensions = 1;
    InitialKeyRepeat = 15;
    "KB_DoubleQuoteOption" = "\\U201cabc\\U201d";
    "KB_SingleQuoteOption" = "\\U2018abc\\U2019";
    "KB_SpellingLanguage" =     {
        "KB_SpellingLanguageIsAutomatic" = 1;
    };
    KeyRepeat = 2;
    NSAutomaticCapitalizationEnabled = 0;
    NSAutomaticPeriodSubstitutionEnabled = 0;
    NSAutomaticSpellingCorrectionEnabled = 0;
    NSLinguisticDataAssetsRequestLastInterval = 86400;
    NSLinguisticDataAssetsRequestTime = "2024-01-21 14:51:34 +0000";
    NSLinguisticDataAssetsRequested =     (
        da,
        en,
        fr,
        nb,
        nn,
        de,
        sv,
        cs,
        nl,
        fi,
        it
    );
    NSLinguisticDataAssetsRequestedByChecker =     (
        da,
        en,
        fr,
        nb,
        nn,
        de,
        sv,
        cs,
        nl,
        fi,
        it
    );`

**Comment by mrnugget** [comment]
> bypasses ApplePressAndHoldEnabled when in VI mode.
> 
> But the Zed terminal does not.

Yeah, that's very interesting. Thanks for digging into this. We'll need to look into this a bit more.
